src/core/include/openvino/core/model.hpp

```diff
@@ -140,10 +140,14 @@ class OPENVINO_API Model : public std::enable_shared_from_this<Model> {
     ov::Output<ov::Node> add_output(const std::string& op_name, size_t output_idx);
     ov::Output<ov::Node> add_output(const ov::Output<ov::Node>& port);
 
-    void reshape(const ov::PartialShape& partial_shape);
-    void reshape(const std::map<size_t, ov::PartialShape>& partial_shapes);
-    void reshape(const std::map<std::string, ov::PartialShape>& partial_shapes);
-    void reshape(const std::map<ov::Output<ov::Node>, ov::PartialShape>& partial_shapes);
+    void reshape(const ov::PartialShape& partial_shape,
+                 const std::unordered_map<std::string, ov::PartialShape>& variable_shapes = {});
+    void reshape(const std::map<size_t, ov::PartialShape>& partial_shapes,
+                 const std::unordered_map<std::string, ov::PartialShape>& variable_shapes = {});
```

Comment:
minor: should it be `map` instead of `unordered_map` to be aligned with 1st argument?

Reply:
I think is not required to have same map type. 
as for inputs shapes the map will be sorted by port number which give some intuition which port will get new shape.
The key for variables is name and if sorted I think is not exactly how they are stored internally. The unordered map will be simpler (faster) for creation.